workers/unity/Packages/com.improbable.gdk.core/Utility/EntityTemplate.cs

```diff
@@ -1,42 +1,272 @@
+        /// <returns>The write access worker attribute, if it exists, null otherwise.</returns>
+        public string GetComponentWriteAccess(uint componentId)
+        {
+            return acl.GetComponentStringAccess(componentId);
+        }
+
+        /// <summary>
+        ///     Retrieves the write access worker attribute for a given component.
+        /// </summary>
+        /// <typeparam name="TSnapshot">The type of the component.</typeparam>
+        /// <returns>The write access worker attribute, if it exists, null otherwise.</returns>
+        public string GetComponentWriteAccess<TSnapshot>() where TSnapshot : struct, ISpatialComponentSnapshot
+        {
+            return GetComponentWriteAccess(DynamicSnapshot.GetSnapshotComponentId<TSnapshot>());
+        }
+
+        /// <summary>
+        ///     Sets the write access worker attribute for a given component.
+        /// </summary>
+        /// <param name="componentId">The component id for that component.</param>
+        /// <param name="writeAccess">The write access worker attribute.</param>
+        public void SetComponentWriteAccess(uint componentId, string writeAccess)
+        {
+            acl.SetComponentWriteAccess(componentId, writeAccess);
+        }
+
+        /// <summary>
+        ///     Sets the write access worker attribute for a given component.
+        /// </summary>
+        /// <param name="writeAccess">The write access worker attribute.</param>
+        /// <typeparam name="TSnapshot">The type of the component.</typeparam>
+        public void SetComponentWriteAccess<TSnapshot>(string writeAccess)
+            where TSnapshot : struct, ISpatialComponentSnapshot
+        {
+            SetComponentWriteAccess(DynamicSnapshot.GetSnapshotComponentId<TSnapshot>(), writeAccess);
+        }
 
-            foreach (var id in template.GetComponentIds())
+        /// <summary>
+        ///     Sets the worker attributes which should have read access over this entity.
+        /// </summary>
+        /// <param name="attributes">The worker attributes which should have read access.</param>
+        public void SetReadAccess(params string[] attributes)
+        {
+            foreach (var attr in attributes)
             {
-                var componentData = template.Get(id);
-                componentData?.SchemaData.Value.Destroy();
+                acl.AddReadAccess(attr);
+            }
+        }
+
+        /// <summary>
+        ///     Creates an <see cref="Entity"/> instance from this template.
+        /// </summary>
+        /// <remarks>
+        ///     This function allocates native memory. The <see cref="Entity"/> returned from this function should
+        ///     be handed to a GDK API, which will take ownership, or otherwise must be disposed of manually.
+        /// </remarks>
+        /// <returns>The Entity object.</returns>
+        public Entity GetEntity()
```

Comment:
Rather than replace things in the template, we could make this a builder that adds the replacements in at this point as overrides.

Reply:
Discussed offline - this can be implemented on top of the existing functionality when we have the use case for it.